cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -888,6 +894,51 @@ public List<ProgramHistory> getRuns(Collection<ProgramId> programs, ProgramRunSt
     });
   }
 
+  @Override
+  public Optional<byte[]> getState(AppStateKey request) {
+    return TransactionRunners.run(transactionRunner, context -> {
```

Comment:
should be `TransactionsRunners.run(transactionRunner, ..., ApplicationNotFoundException.class)` so that the exception gets propagated up as-is instead of wrapped into a runtime exception.